source/dub/project.d

```diff
@@ -1378,18 +1378,20 @@ private string getVariable(Project, Package)(string name, in Project project, in
 
 	if (name == "PLATFORM") {
 		import std.algorithm : filter;
-		foreach (p; gsettings.platform.platform.filter!(p => p != "posix"))
+		import std.range : retro;
+		foreach (p; gsettings.platform.platform[].retro.filter!(p => p != "posix"))
```

Comment:
Could you explain these changes to me?

Reply:
sure, in the `gsettings.platform.platform` Array is stored all platform specifier from the current target, in the least specified order.For example, if you are compiling a project on an Android target, the platform array will contain something like: `[ "posix", "linux", "android" ]`. By reversing this array we get the most specific platform first. We could build the array reversed but I don't know the source code well enough for side effects.

Reply:
oh my mistake, the filter is necessary because in the platform array is `"linux"` before `"posix"`, like:  [ "linux", "posix", "android" ]

Reply:
If I understand what you're saying, and ignoring the [out of the scope of this PR] issue that neither Linux nor Android are technically POSIX compatible nor compliant, this is a bit of a kludge. If possible, assigning a priority (number) to each platform name would be a better (and still simple) solution, then simply take the highest priority (either in place or from some other table).

At very least it needs a comment explaining this code because the use of retro seems like it has the potential to be extremely fragile (Is this ordering actually enforced in dub, I couldn't find anywhere)? If you want to be fancy a DAG could be constructed at compile time specifying the relations between platforms but I wouldn't bother.